timeseries/src/autogluon/timeseries/models/chronos/model.py

```diff
@@ -290,15 +415,12 @@ def _predict(
         # and use that to determine the context length of the model. If the context length is specified
         # during initialization, this is always used. If not, the context length is set to the longest
         # item length. The context length is always capped by self.maximum_context_length.
-        context_length = self.context_length or min(
-            data.num_timesteps_per_item().max(),
-            self.maximum_context_length,
-        )
+        context_length = self._get_context_length(data)
 
         with warning_filter(all_warnings=True):
             import torch
 
-            if self.model_pipeline is None:
+            if self.model_pipeline is None or self.optimization_strategy is not None:
```

Comment:
Why this? Can we make the reason more explicit, also in code? Btw if optimization strategy stuff is too much overhead, we can remove it. We won't be needing CPU acceleration as much anymore.

Reply:
How about this: if fine-tuning is used, we ignore the optimization_strategy? This way the model won't be reloaded every time. Someone who can afford to fine-tune on CPU can probably afford to wait for the slower inference.

Reply:
I am going with ignoring `optimization_strategy` when fine-tuning. Added a FIXME, just in case.